lib/engine/game/g_steam_over_holland/game.rb

```diff
@@ -356,25 +363,42 @@ def non_floated_corporations
         end
 
         def check_distance(route, visits, train = nil)
-          super
-
           raise GameError, 'Route cannot begin/end in a town' if visits.first.town? || visits.last.town?
+
+          super
         end
 
         def revenue_for(route, stops)
-          super
+          revenue = super
+
+          if (ability = abilities(route.corporation, :hex_bonus))
```

Comment:
Looking at the company definitions, there are two companies that have a hex bonus ability. If the same corporation owns both of these then `abilities` will return an array instead of a single ability object.

Reply:
```suggestion
          abilities = Array(abilities(route.corporation, :hex_bonus))
```
This would ensure you always get an array that you could work with.

Reply:
Yes, true! I think I had done it this way because initially I thought each player could only win one private, but that isn't the case. Fixed.